kong/db/init.lua

```diff
@@ -739,30 +739,67 @@ do
                               self.strategy, mig.name)
         end
 
+        if mig.name == "000_base" then
+          local res, err = self.connector:is_014()
+          if err then
+            return nil, fmt_err(self, "unable to detect database version (%s)",
+                                err)
+          end
+
+          if not res.is_eq_014 and not res.is_gt_014 then
+            if res.missing_migration then
+              return nil, fmt_err(self,
+                                  "Migration to 1.0 can only be executed " ..
+                                  "from a fresh or 0.14.x %s %s, but the " ..
+                                  "%s %s is missing some migrations (%s) " ..
+                                  "for %s component, indicating an older " ..
+                                  "version. Migrate to 0.14.x first.",
+                                  self.strategy, self.infos.db_desc,
+                                  self.strategy, self.infos.db_desc,
+                                  res.missing_migration, res.missing_component)
```

Comment:
The messages are still somewhat verbose. Verbose error messages tend to be ignored by users... May I suggest avoiding some repetition (in all 3 of those messages):
```
Migration to 1.0 can only be performed from a 0.14.x postgres database, but
the current one seems to be older (missing migration '%s' for '%s'). Migrate to
0.14.x first, or install 1.0 on a fresh database.
```